src/activities/components/ActivityList.vue

```diff
@@ -355,12 +356,15 @@ export default {
     },
     typeOptions () {
       if (!this.filter) return []
+
+      this.checkArchivedTypes()
```

Comment:
Instead of this kind of imperative style, in vue it is better to use a computed property for this. If you haven't learnt about computed properties yet, maybe https://v2.vuejs.org/v2/guide/computed.html will help.

So, I suggest to create a computed property called `activityTypes` which will run your logic and return the activity types for display in the filter.

Reply:
I'm a little confused by this. I moved the logic into the methods list since we originally had it all in the `typeOptions()` function, but 'yarn lint' threw an error about possible side effects. If we created `activityTypes` as a computed property, what would we do with the return statement on lines 362-374 (where we map each `activityType` in `activityListTypeFilter` to a filter value)?

Reply:
Yes, in a computed function you need to make sure you aren't mutating values, you always return a fresh array with the data, based on the source values.

You would use `this.activityTypes` on line 367, which would be a list containing the right activity types (non-archived + ones in use), calculating in the computed property.

Reply:
Ah I see. I just tried this (creating a computed property, `activityTypes`, with the logic to deal with archived types) but I get the issue again about unintended side effects when calling lint. 

I'm not sure how we can deal with this besides using the `checkArchivedTypes` method.

Reply:
I think I was able to figure it out by the way! I created a new local list in the `activityTypes` computed property and filled in everything from `activityListTypeFilter` so that I could avoid mutating and causing any potential side effects to the data or other fields. Pushing a new commit to address this.